src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -53,13 +53,13 @@ public string Charset
                 HeaderUtilities.ThrowIfReadOnly(IsReadOnly);
                 // We don't prevent a user from setting whitespace-only charsets. Like we can't prevent a user from
                 // setting a non-existing charset.
-                var charsetParameter = NameValueHeaderValue.Find(_parameters, CharsetString);
+                var charsetParameter = NameValueHeaderValue.Find(ParametersInternal, CharsetString);
                 if (string.IsNullOrEmpty(value))
```

Comment:
Put another way, setting `Charset` (or `Boundary`) to `null` should just leave `ParametersInternal` `null` if possible.